### ISSUE TITLE
edns_client_subnet_private should set family

### ISSUE THREAD

**Issue by daareiza** [getdnsapi/getdns#359]
Hi, while playing for a bit with DNS-over-TLS I found an issue using [miekg/dns](https://github.com/miekg/dns) in conjunction with [getdnsapi/stubby](https://github.com/getdnsapi/stubby).

When the configuration option `edns_client_subnet_private` is disabled, all works just as expected, but when is enabled, the new request has some extra EDNS0 flags, including one "empty" subnet that is a "Family 0" address ([code](https://github.com/getdnsapi/getdns/blob/develop/src/stub.c#L152)).

Because of the "Family 0" address, miekg/dns just throws an `dns: bad address family` error while unpacking the EDNS0 message and the request is not resolved. I tried stubby with other TLS resolvers and they work just fine with this "Family 0" address, so I was confused about why miekg/dns has this problem.

Reading the [RFC  7871](https://tools.ietf.org/html/rfc7871#section-7.1.2) specs, I found the following:

```
   If there is no ADDRESS set, i.e.,
   SOURCE PREFIX-LENGTH is set to 0, then FAMILY SHOULD be set to the
   transport over which the query is sent.  This is for
   interoperability; at least one major authoritative server will ignore
   the option if FAMILY is not 1 or 2, even though it is irrelevant if
   there are no ADDRESS bits.
```

So, from the RFC snippet:

- getdns is not respecting the RFC because he is not including the family of the address for the empty subnet
- miekg/dns can be improved by ignoring the EDNS0 subnet option if the Family is not 1 or 2  (I will write an issue there too)

I hope this will help anyone facing similar issues.

**Comment by wtoorop** [comment]
Thank you Daniel.  Acknowledged.

**Comment by wtoorop** [comment]
Should be fixed with commit 6afb02b

**Comment by daareiza** [comment]
thank you so much!!!
